artemis-commons/src/main/java/org/apache/activemq/artemis/core/server/ActiveMQScheduledComponent.java

```diff
@@ -265,6 +291,14 @@ public void stop() {
       if (future != null) {
          future.cancel(false);
          future = null;
+         final AtomicBoolean bookedForRunning = this.bookedForRunning;
+         if (bookedForRunning != null) {
+            // this is trying to fast-fail any running task or pending one:
+            bookedForRunning.compareAndSet(false, true);
```

Comment:
This goes at odds with the prior behaviour, where no attempt was made to stop the tasks already handed to the executor or the final scheduling already being in the process of either handing off another to the executor or running it directly on the scheduledExecutorService. The latter is still true to an extent, the future.cancel(false) remains above, but this new bit will ensure it doesnt actually do anything unless it already passed the gate and starts the 'ActiveMQScheduledComponent.this.run();'.

Have you checked if there is there any impact of this behaviour change? If not, might it be worth omitting this bit, and just leaving the reassignment below for now?

Reply:
> Have you checked if there is there any impact of this behaviour change? If not, might it be worth omitting this bit, and just leaving the reassignment below for now?

It's the safer choice here so will do as you suggest and simplify the rest of the code (although "simplify" is rather optimistic here XD )